packages/functions/transform/src/merge-request-metrics.ts

```diff
@@ -288,65 +288,117 @@ function addUnique(newElement: number, currentArray: number[]) {
   }
 }
 
-async function getId(actorId: number, db: TransformDatabase) {
+type ExtractUserArgs = {
+  externalId: extract.Member['externalId'];
+  name: extract.Member['name'];
+  forgeType: extract.Member['forgeType'];
+  userName: extract.Member['username'];
+}
+
+type TransformUserArgs = {
+  externalId: transform.ForgeUser['externalId'];
+  name: transform.ForgeUser['name'];
+  forgeType: transform.ForgeUser['forgeType'];
+}
+
+async function getExtractUsers(db: ExtractDatabase, userIds: number[]) {
   return await db.select({
-    id: transform.forgeUsers.id,
-  }).from(transform.forgeUsers)
-  .where(eq(transform.forgeUsers.externalId, actorId)).get();
+    externalId: extract.members.externalId,
+    name: extract.members.name,
+    forgeType: extract.members.forgeType,
+    userName: extract.members.username,
+  }).from(extract.members)
+  .where(inArray(extract.members.externalId, userIds))
+  .all();
 }
 
-async function getUserIds(timelineEvents: TimelineEventData[], extractDb: ExtractDatabase, transformDb: TransformDatabase) {
-  const reviewers: number[] = [];
-  const approvers: number[] = [];
-  const committers: number[] = [];
-  let mergedBy;
+async function getExtractUserByName(db: ExtractDatabase, name: string) {
+  return await db.select({
+    externalId: extract.members.externalId,
+    name: extract.members.name,
+    forgeType: extract.members.forgeType,
+  }).from(extract.members)
+  .where(or(
+    eq(extract.members.username, name),
+    eq(extract.members.name, name))
+  ).get()
+}
+
+async function getUserIds(timelineEvents: TimelineEventData[], extractDb: ExtractDatabase, transformDb: TransformDatabase, authorExternalId: number) {
```

Comment:
Generally, I'd prefer to keep our database reads/writes in the giga-chad run function, here you are mixing i/o with processing the timeline (unique set of users), which makes it not (in theory) unit-testable.